### PR TITLE
updated code with ng8 and removed covalent

### PR DESCRIPTION
**Pull request by suprishi**
Adresses/Fixes #<ISSUENO>.

Implements

* 
* 
* 

@devonfw/cobigen

### CODE REVIEW

File: cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-dialog/${variables.etoName#lower_case}-dialog.component.ts.ftl [hunk 1]
```diff
@@ -11,7 +11,7 @@ export class ${variables.etoName?cap_first}DialogComponent {
   title: string;
   items: any = {
   <#list pojo.fields as field>
-    ${field.name}: '',
+    ${field.name?lower_case}: '',
```
Comment (jdiazgon) [inline cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-dialog/${variables.etoName#lower_case}-dialog.component.ts.ftl]: If you use `?lower_case` this means that the whole word will be lower case. Are you sure about this?

For example, imagine that my field is `streetNumber` following camel case standard it should not be converted to `streetnumber`.

I would use `uncap_first` instead.

File: cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-dialog/${variables.etoName#lower_case}-dialog.component.ts.ftl [hunk 2]
```diff
@@ -24,7 +24,11 @@ export class ${variables.etoName?cap_first}DialogComponent {
       this.title = this.getTranslation('${variables.component}.addTitle');
     } else {
       this.title = this.getTranslation('${variables.component}.editTitle');
-      this.items = dialogData;
+      <#list pojo.fields as field>
+        this.items.${field.name?lower_case}: dialogData.${field.name?lower_case};
```
Comment (jdiazgon) [inline cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-dialog/${variables.etoName#lower_case}-dialog.component.ts.ftl]: See my comment about `?lower_case` and `uncap_first`

File: cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-grid/${variables.etoName#lower_case}-grid.component.scss [hunk 3]
```diff
@@ -1,3 +1,51 @@
-.td-expansion-panel-header {
-    background-color: #cccccc;
+mat-card-title {
+  padding-top: 16px;
+  padding-left: 16px;
+  padding-right: 16px;
```
Comment (jdiazgon) [inline cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-grid/${variables.etoName#lower_case}-grid.component.scss]: Maybe there is a reason behind this, but pixels are usually not a good idea. Depending on the screen density, it may look different. 

I think we should use percentages, do you agree @sjimenez77 ?

File: cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-grid/${variables.etoName#lower_case}-grid.component.ts.ftl [hunk 4]
```diff
@@ -1,70 +1,67 @@
...
+  pagingBar: MatPaginator;
 
   data: any = [];
-  columns: ITdDataTableColumn[] = [
+  columns: any[] = [
   <#list pojo.fields as field>
     {
-      name: '${field.name}',
-      label: this.getTranslation('${variables.component?lower_case}.${variables.etoName?cap_first}.columns.${field.name}'),
+      name: '${field.name?lower_case}',
```
Comment (jdiazgon) [inline cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-grid/${variables.etoName#lower_case}-grid.component.ts.ftl]: See my comment about `?lower_case` and `uncap_first`

File: cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-grid/${variables.etoName#lower_case}-grid.component.ts.ftl [hunk 5]
```diff
@@ -83,20 +80,22 @@ export class ${variables.etoName?cap_first}GridComponent implements OnInit {
...
-          this.dataTable.refresh();
         },
         (error: any) => {
           setTimeout(() => {
-            this._dialogService.openAlert({
-              message: error.message,
-              title: this.getTranslation('ERROR'),
-              closeButton: 'CLOSE',
+            this.dialog.open(${variables.etoName?cap_first}AlertComponent, {
+              width: '400px',
```
Comment (jdiazgon) [inline cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-grid/${variables.etoName#lower_case}-grid.component.ts.ftl]: Again here we should be careful with pixels.

File: cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/services/${variables.etoName#lower_case}.service.ts.ftl [hunk 6]
```diff
@@ -17,7 +14,7 @@ export class ${variables.etoName?cap_first}Service {
     size: number,
     page: number,
     searchTerms: any,
-    sort: Sort[],
+    sort: any[],
```
Comment (jdiazgon) [inline cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/services/${variables.etoName#lower_case}.service.ts.ftl]: Why the change from `Sort[]` to `any[]`?

File: cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/services/${variables.etoName#lower_case}.service.ts.ftl [hunk 7]
```diff
@@ -26,7 +23,7 @@ export class ${variables.etoName?cap_first}Service {
         sort: sort,
       },
     <#list pojo.fields as field>
-      ${field.name}: searchTerms.${field.name},
+      ${field.name?lower_case}: searchTerms.${field.name?lower_case},
```
Comment (jdiazgon) [inline cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/services/${variables.etoName#lower_case}.service.ts.ftl]: See my comment about `?lower_case` and `uncap_first`

File: cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-dialog/${variables.etoName#lower_case}-dialog.component.ts.ftl (hunk 1 above)
Comment (suprishi) [reply cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-dialog/${variables.etoName#lower_case}-dialog.component.ts.ftl]: One of the fields in the application is `AGE`. Using `uncap_first` makes it `aGE`. How about if I chain `capitalize` and `uncap_first` together?

File: cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-grid/${variables.etoName#lower_case}-grid.component.scss (hunk 3 above)
Comment (suprishi) [reply cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-grid/${variables.etoName#lower_case}-grid.component.scss]: I checked it in a few of the devices in Chrome's device emulator. Visually it looks alright.

If we switch to percentages we need to calculate the correct percentage for different screen dimensions and use media queries to implement it. Seems too much work for a small thing. Should I proceed anyhow?

File: cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-grid/${variables.etoName#lower_case}-grid.component.ts.ftl (hunk 5 above)
Comment (suprishi) [reply cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-grid/${variables.etoName#lower_case}-grid.component.ts.ftl]: This one is applied only on larger devices. It gets a `max-width` style of `80vw` applied by Angular cdk layout so it looks fine on smaller devices as well.

File: cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-dialog/${variables.etoName#lower_case}-dialog.component.ts.ftl (hunk 1 above)
Comment (jdiazgon) [reply cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-dialog/${variables.etoName#lower_case}-dialog.component.ts.ftl]: > One of the fields in the application is AGE

This is the problem that caused the back-end to fail. If a user defines a Java Entity, and it does not comply with naming conventions (it should be `age`), then we should not fix his mistakes.

Our developers need to follow naming conventions, no one should define a field as `AGE`.

File: cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-grid/${variables.etoName#lower_case}-grid.component.scss (hunk 3 above)
Comment (jdiazgon) [reply cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-grid/${variables.etoName#lower_case}-grid.component.scss]: I see. I think that as we are on a hurry we should proceed without fixing it, it is a nice to have feature.

File: cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-grid/${variables.etoName#lower_case}-grid.component.ts.ftl (hunk 5 above)
Comment (jdiazgon) [reply cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-grid/${variables.etoName#lower_case}-grid.component.ts.ftl]: Okay, thanks for the explanation.

File: cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-dialog/${variables.etoName#lower_case}-dialog.component.ts.ftl (hunk 1 above)
Comment (suprishi) [reply cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-dialog/${variables.etoName#lower_case}-dialog.component.ts.ftl]: So I'll proceed with replacing `lower_case` with `uncap_first` wherever required. Correct?

File: cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/services/${variables.etoName#lower_case}.service.ts.ftl (hunk 6 above)
Comment (suprishi) [reply cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/services/${variables.etoName#lower_case}.service.ts.ftl]: Looks like a mistake. Will correct it in the next commit.

File: cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-dialog/${variables.etoName#lower_case}-dialog.component.ts.ftl (hunk 1 above)
Comment (jdiazgon) [reply cobigen-templates/templates-devon4j/src/main/templates/crud_angular_client_app/templates/src/app/${variables.etoName#lower_case}/${variables.etoName#lower_case}-dialog/${variables.etoName#lower_case}-dialog.component.ts.ftl]: Correct

File: cobigen-templates/templates-devon4j/src/main/templates/crud_ngrx_client_app/templates.xml [hunk 8]
```diff
@@ -0,0 +1,82 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<templatesConfiguration xmlns="http://capgemini.com/devonfw/cobigen/TemplatesConfiguration"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="2.1">
+
+  <templates>
+    <templateExtension ref="${variables.etoName#lower_case}-dialog.component.ts" mergeStrategy="tsmerge"/>
+    <templateExtension ref="${variables.etoName#lower_case}-grid.component.ts" mergeStrategy="tsmerge"/>
```
Comment (jdiazgon) [inline cobigen-templates/templates-devon4j/src/main/templates/crud_ngrx_client_app/templates.xml]: Can you change all the `mergeStrategy` to `override` for TypeScript files? Html or Json can stay as they are :)

File: cobigen-templates/templates-devon4j/src/main/templates/crud_ngrx_client_app/templates.xml [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+    <templateExtension ref="home.component.html" mergeStrategy="override"/>
+    <templateExtension ref="${variables.etoName#lower_case}.model.ts" mergeStrategy="tsmerge"/>
+    <templateExtension ref="datadetailstest.model.ts" mergeStrategy="tsmerge"/>
+    <templateExtension ref="httpresponse.model.ts" mergeStrategy="tsmerge"/>
+    <templateExtension ref="searchcriteriadata.model.ts" mergeStrategy="tsmerge"/>
+    <templateExtension ref="${variables.etoName#lower_case}.actions.ts" mergeStrategy="tsmerge"/>
+    <templateExtension ref="${variables.etoName#lower_case}.effects.ts" mergeStrategy="tsmerge"/>
+    <templateExtension ref="${variables.etoName#lower_case}.reducers.ts" mergeStrategy="tsmerge"/>
+    <templateExtension ref="${variables.etoName#lower_case}.selectors.ts" mergeStrategy="tsmerge"/>
+    <templateExtension ref="actions_index.ts" destinationPath="../../../workspaces_vs/devon4ng-ngrx-cobigen-template/src/app/${variables.etoName#lower_case}/store/actions/index.ts" mergeStrategy="tsmerge"/>
```
Comment (jdiazgon) [inline cobigen-templates/templates-devon4j/src/main/templates/crud_ngrx_client_app/templates.xml]: `destinationPath` should be the parent folder where the input file is located. For instance, if I have a devon4j project in `foo\devon4j\...\inputFile`, then the code should be generated to `foo\devon4ng-ngrx\...`

Please change the destination paths. For devon4ng is `relocate:../../devon4ng-application-template/${cwd}`

File: cobigen-templates/templates-devon4j/src/main/templates/crud_ngrx_client_app/templates/cobigen.properties [hunk 10]
```diff
@@ -0,0 +1 @@
+relocate:../../../workspaces_vs/devon4ng-ngrx-cobigen-template/${cwd}
```
Comment (jdiazgon) [inline cobigen-templates/templates-devon4j/src/main/templates/crud_ngrx_client_app/templates/cobigen.properties]: Please change the destination paths. For devon4ng is `relocate:../../devon4ng-application-template/${cwd}`

Comment (suprishi) [reply cobigen-templates/templates-devon4j/src/main/templates/crud_ngrx_client_app/templates/cobigen.properties]: Shouldn't it be `relocate:../../devon4ng-ngrx-template/${cwd}` since we are using the [devon4ng-ngrx-template](https://github.com/devonfw/devon4ng-ngrx-template) as the base template?

Comment (jdiazgon) [reply cobigen-templates/templates-devon4j/src/main/templates/crud_ngrx_client_app/templates/cobigen.properties]: Yes, sorry, it was an example. You are right.
